### PR TITLE
Python2 removal.

### PR DESCRIPTION
**Pull request by gbin**
A lot of crap all around: try on imports / if PY2/Py3, tests on unicode,
src conversion, travis config, docs ...

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -11,8 +10,7 @@ before_script:
 
 install:
   - pip install --upgrade pip
-  - pip install -e .
-  - pip install slackclient  # Optional dependency but Slack tests are skipped without it
+  - pip install -e .[slack]
```
Comment (zoni) [inline .travis.yml]: I would leave this the way it was, I think that's a little more clear (and the comment itself is important as well)

**Comment by zoni** [comment]
Aside from the comment above, this looks good in sofar as I can tell.

File: docs/user_guide/plugin_development/basics.rst [hunk 2]
```diff
@@ -140,7 +140,7 @@ Lets go ahead and create ours. Place the following in a file called
     file structure.
 
 Lets look at what this does. We see three sections, `[Core]` ,
-`[Python]` and `[Documentation]`. The `[Core]` section is what tells
+and `[Documentation]`. The `[Core]` section is what tells
```
Comment (Ian-Foote) [inline docs/user_guide/plugin_development/basics.rst]:
```
Lets look at what this does. We see two sections, `[Core]` 
and `[Documentation]`. The `[Core]` section is what tells
```

File: .travis.yml (hunk 1 above)
Comment (gbin) [inline .travis.yml]: I was thinking, maybe another simple test doing `pip install -e .[all]` might be useful to see if the entire dependency tree is still buildable.

**Pull request merged**